.github/workflows/test-python.yml

```diff
@@ -67,7 +67,11 @@ jobs:
           mypy --install-types --non-interactive bson/codec_options.py
           mypy --install-types --non-interactive --disable-error-code var-annotated --disable-error-code attr-defined --disable-error-code union-attr --disable-error-code assignment --disable-error-code no-redef --disable-error-code index --allow-redefinition --allow-untyped-globals --exclude "test/mypy_fails/*.*" test
           python -m pip install -U typing_extensions
-          mypy --install-types --non-interactive test/test_mypy.py
+          mypy --install-types --non-interactive test/test_typing.py test/test_typing_strict.py
+          mypy --strict test/test_typing_strict.py
+          pyright test/test_typing.py test/test_typing_strict.py
+          echo '{"strict": ["tests/test_typing_strict.py"]}"'' >>  pyrightconfig.json
+          pyright test/test_typing_strict.py
```

Comment:
Can we split these into different steps?
```
      - name: Run mypy
        run: |
          mypy --install-types --non-interactive bson gridfs tools pymongo
          ...
      - name: Run mypy --strict
        run: |
          ...
      - name: Run pyright
        run: |
          ...
      - name: Run pyright strict mode
        run: |
          ...
```

Reply:
Done